File: pytools/data/data.code.cpp.server/MsgTestXX.cpp
#ifndef _MSG_TEST_X_X_
#define _MSG_TEST_X_X_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class MsgTestXX
{
private:
	U8 m_id_u8;
	list<F32> m_id_f32;
	U8 m_id_op_u8_flag;
	U8 m_id_op_u8;


public:
	BYTE* Encode(U32* len)
	{
		Packet packet;
		packet.WriteU8(m_id_u8);
		U16 id_f32_count = (U16)m_id_f32.size();
		packet.WriteU16(id_f32_count);
		for (list<F32>::iterator i = m_id_f32.begin(); i != m_id_f32.end(); i++)
		{
			packet.WriteF32((F32)*i);
		}
		packet.WriteU8(m_id_op_u8_flag);
		if (m_id_op_u8_flag == 1)
		{
			packet.WriteU8(m_id_op_u8);
		}
		return packet.ReadBytes(len);
	}

	MsgTestXX()
	{

	}

	MsgTestXX(Packet* packet)
	{
		m_id_u8 = packet->ReadU8();
		U16 id_f32_count = packet->ReadU16();
		for (U16 i = 0; i < id_f32_count; i++)
		{
			m_id_f32.push_back(packet->ReadF32());
		}
		m_id_op_u8_flag = packet->ReadU8();
		if (m_id_op_u8_flag == 1)
		{
			m_id_op_u8 = packet->ReadU8();
		}
	}

	BYTE* GetBytes(U32* len)
	{
		return Encode(len);
	}


	void SetIdU8(U8 id_u8)
	{
		m_id_u8 = id_u8;
	}
	U8 GetIdU8()
	{
		return m_id_u8;
	}

	void SetIdF32(list<F32> id_f32)
	{
		m_id_f32 = id_f32;
	}
	list<F32> GetIdF32()
	{
		return m_id_f32;
	}

	void SetIdOpU8(U8 id_op_u8)
	{
		m_id_op_u8_flag = 1;
		m_id_op_u8 = id_op_u8;
	}
	U8 GetIdOpU8()
	{
		return m_id_op_u8;
	}

};

#endif

File: pytools/data/data.code.golang.client/msg_test_x_x.go
package proto

import (
	"packet"
)

type MsgTestXX struct {
	idU8                     uint8
	idF32                    []float32
	idOpU8Flag               uint8
	idOpU8                   uint8
}

func (this *MsgTestXX) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint8(this.idU8)
	idF32Count := uint16(len(this.idF32))
	pack.WriteUint16(idF32Count)
	for i := uint16(0); i < idF32Count; i++ {
		pack.WriteFloat32(this.idF32[i])
	}
	pack.WriteUint8(this.idOpU8Flag)
	if this.idOpU8Flag == 1 {
		pack.WriteUint8(this.idOpU8)
	}

	return pack.ReadBytes()
}

func MsgTestXXDecode(pack *packet.Packet) *MsgTestXX {
	msgTestXX := &MsgTestXX{}

	msgTestXX.idU8 = pack.ReadUint8()
	idF32Count := pack.ReadUint16()
	for ;idF32Count > 0; idF32Count-- {
		msgTestXX.idF32 = append(msgTestXX.idF32, pack.ReadFloat32())
	}
	msgTestXX.idOpU8Flag = pack.ReadUint8()
	if msgTestXX.idOpU8Flag == 1 {
		msgTestXX.idOpU8 = pack.ReadUint8()
	}
	return msgTestXX
}

func (this *MsgTestXX) SetIdU8(idU8 uint8) {
	this.idU8 = idU8
}

func (this *MsgTestXX) GetIdU8() uint8 {
	return this.idU8
}

func (this *MsgTestXX) SetIdF32(idF32 []float32) {
	this.idF32 = idF32
}

func (this *MsgTestXX) GetIdF32() []float32 {
	return this.idF32
}

func (this *MsgTestXX) SetIdOpU8(idOpU8 uint8) {
	this.idOpU8Flag = 1
	this.idOpU8 = idOpU8
}

func (this *MsgTestXX) GetIdOpU8() uint8 {
	return this.idOpU8
}
